 	local display_reset = (not flow.hideresetbtn)
 
 	if pageaction and
-	   (display_back or display_skip or display_apply or display_save or display_reset)
+	   (display_back or display_skip or display_last or display_next or display_apply or display_save or display_reset)
 	then
 		%><div class="cbi-page-actions"><%
 
-		if display_back then
+		if display_back then 
 			%><input class="cbi-button cbi-button-link" type="button" value="<%:Back to Overview%>" onclick="location.href='<%=pcdata(redirect)%>'" /> <%
 		end
 
@@ -18,6 +20,14 @@
 			%><input class="cbi-button cbi-button-skip" type="button" value="<%:Skip%>" onclick="cbi_submit(this, 'cbi.skip')" /> <%
 		end
 
+		if display_last then
+			%><input class="cbi-button cbi-button-save" type="button" value="<%:Last%>" onclick="location.href='<%=pcdata(flow.last)%>'" /> <%
+		end
+		
+		if display_next then
+			%><input class="cbi-button cbi-button-apply" type="submit" value="<%:Next%>" /> <%
+		end
+		
 		if display_apply then
 			%><input class="cbi-button cbi-button-apply" type="button" value="<%:Save & Apply%>" onclick="cbi_submit(this, 'cbi.apply')" /> <%
 		end
diff --git a/themes/luci-theme-material/htdocs/luci-static/material/cascade.css b/themes/luci-theme-material/htdocs/luci-static/material/cascade.css
index 4875c33..4fc85ed 100644
--- a/themes/luci-theme-material/htdocs/luci-static/material/cascade.css
+++ b/themes/luci-theme-material/htdocs/luci-static/material/cascade.css
@@ -421,7 +421,7 @@ header > .fill > .container > .brand {
     background-color: #0099CC;
 }
 
-.main > .main-left > .nav > .slide > .slide-menu > li > a {
+.main > .main-left > .nav > .slide > .slide-menu > li > a,.main > .main-left > .nav  > li > a {
     text-decoration: none;
     white-space: nowrap;
 }
